### PR TITLE
feat: adds support for env var replacement for auth policy values

### PR DESCRIPTION
**Pull request by jajeffries**
This pull request introduces environment variable substitution for SNMP authentication fields, refactors the code to centralize environment variable resolution logic, and adds comprehensive tests for the new functionality. The changes improve configurability, code reuse, and reliability.

### Enhancements to SNMP Authentication Configuration:
* Updated `snmp-discovery/README.md` to document environment variable substitution for SNMP authentication fields (`community`, `username`, `auth_passphrase`, `priv_passphrase`) and added an example configuration. If an environment variable is not set, the service exits with an error.

### Code Refactoring:
* Removed the `resolveEnv` function from `snmp-discovery/cmd/main.go` and replaced its usage with the new utility function `utils.ResolveEnvOrExit`. [[1]](diffhunk://#diff-08cb12983d08e9d6370e03e2761fc8d686dba3469fd5914ef1a912471b482139L10) [[2]](diffhunk://#diff-08cb12983d08e9d6370e03e2761fc8d686dba3469fd5914ef1a912471b482139L87-R78)
* Added `resolveAuthenticationEnvVars` to `snmp-discovery/policy/manager.go` to handle environment variable resolution for authentication fields in policies.

### Utility Functions:
* Introduced `ResolveEnv` and `ResolveEnvOrExit` utility functions in `snmp-discovery/utils/utils.go` for resolving environment variables, with error handling and process termination for unset variables.

### Unit Tests:
* Added comprehensive tests in `snmp-discovery/utils/utils_test.go` for `ResolveEnv` and `ResolveEnvOrExit`, covering various edge cases such as unset variables, partial syntax, and mixed content.
* Extended tests in `snmp-discovery/policy/manager_test.go` to validate environment variable resolution in policies, including scenarios with mixed values and multiple policies.

### CODE REVIEW

File: snmp-discovery/policy/manager.go [hunk 1]
```diff
@@ -213,3 +215,22 @@ func (m *Manager) Stop() error {
...
+		auth.Community = resolved
+	}
+	if resolved, err := utils.ResolveEnv(auth.Username); err == nil {
+		auth.Username = resolved
+	}
+	if resolved, err := utils.ResolveEnv(auth.AuthPassphrase); err == nil {
+		auth.AuthPassphrase = resolved
+	}
+	if resolved, err := utils.ResolveEnv(auth.PrivPassphrase); err == nil {
+		auth.PrivPassphrase = resolved
```
[the remark this answers is not in the document]
Comment (jajeffries) [reply snmp-discovery/policy/manager.go]: Fixed

File: snmp-discovery/utils/utils.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	if strings.HasPrefix(value, "${") && strings.HasSuffix(value, "}") {
+		// Extract the environment variable name
+		envVar := value[2 : len(value)-1]
+		// Skip empty environment variable names
+		if envVar == "" {
+			return value, nil
+		}
+		// Get the value of the environment variable
+		envValue := os.Getenv(envVar)
+		if envValue != "" {
```
[the remark this answers is not in the document]
Comment (jajeffries) [reply snmp-discovery/utils/utils.go:24]: Unnecessary

File: snmp-discovery/policy/manager.go [hunk 3]
```diff
@@ -213,3 +217,35 @@ func (m *Manager) Stop() error {
...
+		return fmt.Errorf("failed to resolve auth_passphrase environment variable: %w", err)
+	}
+	auth.AuthPassphrase = resolved
+
+	resolved, err = utils.ResolveEnv(auth.PrivPassphrase)
+	if err != nil {
+		return fmt.Errorf("failed to resolve priv_passphrase environment variable: %w", err)
+	}
+	auth.PrivPassphrase = resolved
+
```
[the remark this answers is not in the document]
Comment (jajeffries) [reply snmp-discovery/policy/manager.go]: Applied

**Review by leoparente — APPROVED** [review]
